python_modules/dagster/dagster/_core/definitions/repository_definition/repository_definition.py

```diff
@@ -48,26 +48,36 @@
     from dagster._core.storage.asset_value_loader import AssetValueLoader
 
 
-@whitelist_for_serdes
+@whitelist_for_serdes(
+    storage_field_names={"cacheable_asset_data": "cached_data_by_key"},
+)
 class RepositoryLoadData(
     NamedTuple(
         "_RepositoryLoadData",
         [
-            ("cached_data_by_key", Mapping[str, Sequence[AssetsDefinitionCacheableData]]),
+            ("cacheable_asset_data", Mapping[str, Sequence[AssetsDefinitionCacheableData]]),
```

Comment:
I don't see enough upside here to warrant a rename and dealing with storage back compat

Reply:
Storage back compat is solved by the `storage_field_names` param above-- basically this is always serialized as/deserialized from the old name `cached_data_by_key`.

`cached_data_by_key` becomes very confusing if we have another field on here containing cached data.

Reply:
We should not used "cached" in my view in the verbiage at all.

That all being said I don't care enough to block this PR on this issue.

Reply:
>We should not used "cached" in my view in the verbiage at all.

The rename gets rid of "cached" so hopefully that's an improvement then. "cacheable" in the new name `cacheable_asset_data` is just to very explicitly link this data to its source in `CacheableAssetsDefinition`